tests/integration/self_update_test.sh

```diff
@@ -0,0 +1,213 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './src/lib/kwlib.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -a KW_BRANCHES=('master' 'unstable')
+declare -g HOST_KW_CLONED_PATH
```

Comment:
`KW_BRANCHES` can be read-only with -ar. However, `HOST_KW_CLONED_PATH` needs to be assigned a value later, so it can't be read-only (-gr).

Also, combining declaration and assignment triggers a shellcheck `SC2155` error.